src/V3Randomize.cpp

```diff
@@ -696,7 +921,13 @@ class ConstraintExprVisitor final : public VNVisitor {
             AstVarRef* const varRefp
                 = new AstVarRef{varp->fileline(), classp, varp, VAccess::WRITE};
             varRefp->classOrPackagep(classOrPackagep);
-            membersel ? methodp->addPinsp(membersel) : methodp->addPinsp(varRefp);
+            if (membersel) {
+                methodp->addPinsp(membersel);
+                // Delete the unused varRefp to avoid memory leak
+                VL_DO_DANGLING(varRefp->deleteTree(), varRefp);
+            } else {
+                methodp->addPinsp(varRefp);
+            }
```

Comment:
Avoid unnecessary temporary allocations
```suggestion
            if (membersel) {
                methodp->addPinsp(membersel);
            } else {
                AstVarRef* const varRefp
                    = new AstVarRef{varp->fileline(), classp, varp, VAccess::WRITE};
                varRefp->classOrPackagep(classOrPackagep);
                methodp->addPinsp(varRefp);
            }
```